test/adaptive_load/metrics_evaluator_test.cc

```diff
@@ -0,0 +1,399 @@
+// #include "api/adaptive_load/input_variable_setter_impl.pb.h"
+#include "api/adaptive_load/metric_spec.pb.h"
+#include "api/adaptive_load/metrics_plugin_impl.pb.h"
+#include "api/adaptive_load/scoring_function_impl.pb.h"
+// #include "api/adaptive_load/step_controller_impl.pb.h"
+#include "api/client/options.pb.h"
+#include "api/client/output.pb.h"
+#include "api/client/service.grpc.pb.h"
+#include "api/client/service.pb.h"
+
+#include "test/adaptive_load/fake_plugins/fake_metrics_plugin/fake_metrics_plugin.h"
+#include "test/adaptive_load/minimal_output.h"
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/status/status.h"
+#include "absl/strings/str_join.h"
+#include "adaptive_load/metrics_plugin_impl.h"
+#include "adaptive_load/plugin_loader.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+namespace {
+
+using ::Envoy::Protobuf::util::MessageDifferencer;
+using ::nighthawk::adaptive_load::BenchmarkResult;
+using ::nighthawk::adaptive_load::FakeMetricsPluginConfig;
+using ::nighthawk::adaptive_load::MetricEvaluation;
+using ::nighthawk::adaptive_load::MetricSpec;
+using ::nighthawk::adaptive_load::ThresholdSpec;
+using ::testing::HasSubstr;
+
+/**
+ * Creates a valid TypedExtensionConfig proto selecting the real BinaryScoringFunction plugin
+ * and configuring it with a threshold.
+ *
+ * @param lower_threshold Threshold value to set within the config proto.
+ *
+ * @return TypedExtensionConfig Full scoring function plugin spec that selects
+ * nighthawk.binary_scoring and provides a config.
+ */
+envoy::config::core::v3::TypedExtensionConfig
+MakeLowerThresholdBinaryScoringFunctionConfig(double lower_threshold) {
+  envoy::config::core::v3::TypedExtensionConfig config;
+  config.set_name("nighthawk.binary_scoring");
+  nighthawk::adaptive_load::BinaryScoringFunctionConfig inner_config;
+  inner_config.mutable_lower_threshold()->set_value(lower_threshold);
+  config.mutable_typed_config()->PackFrom(inner_config);
+  return config;
+}
+
+/**
+ * Creates a simulated Nighthawk Service response that reflects the specified send rate.
+ *
+ * @param send_rate The send rate that the proto values should represent.
+ *
+ * @return ExecutionResponse A simulated Nighthawk Service response with counters representing the
+ * specified send rate, along with other dummy counters and stats.
+ */
+nighthawk::client::ExecutionResponse MakeNighthawkResponseWithSendRate(double send_rate) {
+  nighthawk::client::ExecutionResponse response;
+  nighthawk::client::Output output = MakeSimpleNighthawkOutput({
+      /*concurrency=*/"auto",
+      /*requests_per_second=*/1024,
+      /*actual_duration_seconds=*/10,
+      /*upstream_rq_total=*/static_cast<int>(10 * 1024 * send_rate),
+      /*response_count_2xx=*/320,
+      /*min_ns=*/400,
+      /*mean_ns=*/500,
+      /*max_ns=*/600,
+      /*pstdev_ns=*/11,
+  });
+  *response.mutable_output() = output;
+  return response;
+}
+
+TEST(EvaluateMetric, SetsMetricId) {
```

Comment:
I think we prefer the ConditionAWillResultinB format for naming these.

Reply:
Added some more detail throughout.